### PR TITLE
Check for allowed work package context menu action

### PR DESCRIPTION
**Pull request by bluemoonecho**
At the moment this PR is adding a `isAllowedActions` helper and filtering `possible actions` into `permitted actions`

### CODE REVIEW

File: frontend/src/app/features/work-packages/components/wp-table/context-menu-helper/wp-context-menu-helper.service.ts [hunk 1]
```diff
@@ -103,13 +103,16 @@ export class WorkPackageContextMenuHelperService {
 
   public getIntersectOfPermittedActions(workPackages:any) {
     const bulkPermittedActions:any = [];
+
```
Comment (tessi) [inline frontend/src/app/features/work-packages/components/wp-table/context-menu-helper/wp-context-menu-helper.service.ts]:
```suggestion
```

File: frontend/src/app/features/work-packages/components/wp-table/context-menu-helper/wp-context-menu-helper.service.ts [hunk 2]
```diff
@@ -103,13 +103,16 @@ export class WorkPackageContextMenuHelperService {
...
     const possibleActions = this.BULK_ACTIONS.concat(this.HookService.call('workPackageBulkContextMenu'));
-    const permittedActions = _.filter(possibleActions, (action:any) => _.every(workPackages, (workPackage:WorkPackageResource) => this.getAllowedActions(workPackage, [action]).length >= 1));
+    const permittedActions = _.filter(possibleActions, (action:any) => _.every(workPackages, (workPackage:WorkPackageResource) => this.isActionAllowed(workPackage, action)));
```
Comment (tessi) [inline frontend/src/app/features/work-packages/components/wp-table/context-menu-helper/wp-context-menu-helper.service.ts:107]: wait, shouldn't this be part of the previous pr/commit/branch?

File: frontend/src/app/features/work-packages/components/wp-table/context-menu-helper/wp-context-menu-helper.service.ts [hunk 3]
```diff
@@ -103,13 +103,16 @@ export class WorkPackageContextMenuHelperService {
...
 
     _.each(permittedActions, (permittedAction:any) => {
       bulkPermittedActions.push({
         key: permittedAction.key,
         text: permittedAction.text,
+        // the icon field is necessary for custom plugins
```
Comment (tessi) [inline frontend/src/app/features/work-packages/components/wp-table/context-menu-helper/wp-context-menu-helper.service.ts]: I like this comment as a thought, but don't think it has much additional value being distributed in the code
so let's keep it here and remove it from the source code :)
```suggestion
```

File: frontend/src/app/features/work-packages/components/wp-table/context-menu-helper/wp-context-menu-helper.service.ts (hunk 2 above)
Comment (tessi) [reply frontend/src/app/features/work-packages/components/wp-table/context-menu-helper/wp-context-menu-helper.service.ts:107]: ah wait, this is the PR about fixing the permitted actions? Don't we have that already sitting as a PR on opf/openproject?

File: frontend/src/app/features/work-packages/components/wp-table/context-menu-helper/wp-context-menu-helper.service.ts [hunk 4]
```diff
@@ -103,13 +103,16 @@ export class WorkPackageContextMenuHelperService {
...
+        icon: permittedAction.icon,
```
Comment (tessi) [inline frontend/src/app/features/work-packages/components/wp-table/context-menu-helper/wp-context-menu-helper.service.ts]: if this is the PR about the shown actions, I believe we don't need that change here. I imagine this should be in the PR where we introduce the new configurable bulk actions, right?

File: frontend/src/app/features/work-packages/components/wp-table/context-menu-helper/wp-context-menu-helper.service.ts (hunk 2 above)
Comment (bluemoonecho) [reply frontend/src/app/features/work-packages/components/wp-table/context-menu-helper/wp-context-menu-helper.service.ts:107]: Oh I combined both of them, but maybe it was not a good idea.
I will change the title of this Pr, revert it to only take care of the allowed actions check, and maybe add the further `icon` changes to your branch, if that's ok?

Comment (bluemoonecho) [reply frontend/src/app/features/work-packages/components/wp-table/context-menu-helper/wp-context-menu-helper.service.ts:107]: At the moment this PR is pointing to the "Allow plugins to link into the WP bulk context menu".
But if we get the one on the Open Project repo merged, we won't need this one.

**Pull request closed without merging**